documentation/hacker-guide/source/runtime/startup.rst

```diff
@@ -0,0 +1,138 @@
+Startup
+=======
+
+When a Dylan library (either a shared library/DLL or executable) is
+loaded, a number of initialization tasks need to take place:
+
+- The Dylan run-time facilities, including the garbage collector and
+  the main thread's thread environment block (TEB), need to be
+  initialized if they haven't already been.
+- The final addresses of any :drm:`<symbol>` objects referenced by the
+  library need to be resolved, and data locations that reference
+  symbols need to be patched to point to the resolved addresses. This
+  is part of the "for-system" initialization.
+- Top-level forms in each of the library's source files need to be
+  evaluated. This is the "for-user" initialization.
+
+The following describes how Dylan libraries are initialized under the
+different compiler back-ends.
+
+LLVM
+----
+
+Initialization for programs compiled by the LLVM back-end relies on
+constructor functions, provided on most platforms to support (among
+other things) C++ constructors. The LLVM linker makes use of two
+different constructor priorities, system (priority 0) and user
+(priority 65535).
+
+The first constructor generated at system priority within the
+``_glue.bc`` file generated for the ``dylan`` library is a call to
+``_Init_Run_Time``, a C function defined in
+``sources/lib/run-time/llvm-runtime-init.c`` to initialize the garbage
+collector and other system facilities needed by the Dylan run-time.
+
+For each compile unit that requires it, a constructor at system
+priority is generated to do for-system initialization:
+
+- When :drm:`<symbol>` resolution is required, a call to
+  ``primitive-symbol-fixup`` is generated, passing a table of symbols
+  to resolve and reference addresses to patch.
+
+- If there are any system initializers in the compile unit, calls to
+  these are also generated. These normally only occur when a class
+  definition depends on non-static values.
+
+The remaining for-user initialization is done in the per-library
+startup glue function, named ``_Init_``\ *mangled-library-name*, where
```

Comment:
There are a variety of ``\`` scattered through the file ... are they for anything or a remnant from something else?